src/IssueViz/Editor/QuickActions/DeselectIssueVisualizationAction.cs

```diff
@@ -0,0 +1,49 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Threading;
+using SonarLint.VisualStudio.IssueVisualization.Selection;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.QuickActions
+{
+    internal class DeselectIssueVisualizationAction : BaseSuggestedAction
+    {
+        private readonly IAnalysisIssueSelectionService selectionService;
+
+        /// <summary>
+        /// <see cref="IssueLocationActionsSource.GetSuggestedActions"/> can return a cached version of the actions, which will show a deselect action even though the issue has already been deselected.
```

Comment:
It's not clear what this means. Does it mean a lightbulb for the deselect action will be shown when it shouldn't be?

Reply:
Yes. In general, GetSuggestedActions can return cached actions. For example:
1. Place the caret on span that should have a lightbulb
2. Open the lightbulb with the keyboard shortcut
3. Select an action ==> action works
4. Don't move the caret, and open the lightbulb again with the keyboard shortcut ==> GetSuggestedActions won't be called. The list will be populated with previous items. If move the caret, GetSuggestedActions is called.
Registering to TagsChanged should've fixed that, as it raises the SuggestedActionsChanged event. But it doesn't. SuggestedActionsChanged is indeed raised, but GetSuggestedActions is still not being called.

Reply:
Btw, unrelated: do you think it's worth uncapitalizing the text? e.g. `cpp:S3776: hide SonarLint issue visualization` ?

Reply:
Caps - look at what other suggestions do and go with whatever is most common.